test/e2e/httpproxy/exact_path_condition_match_test.go

```diff
@@ -0,0 +1,158 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package httpproxy
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func testExactPathCondition(namespace string) {
+	Specify("Exact path match routing works and has precedence over prefix match", func() {
+		var (
+			t                = f.T()
+			serviceNamespace = namespace
+		)
+
+		f.Fixtures.Echo.Deploy(serviceNamespace, "echo-blue")
+		f.Fixtures.Echo.Deploy(serviceNamespace, "echo-green")
+		f.Fixtures.Echo.Deploy(serviceNamespace, "echo-default")
+
+		serviceProxy := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: serviceNamespace,
+				Name:      "echo-exact",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				VirtualHost: &contourv1.VirtualHost{
+					Fqdn: "exactpathcondition.projectcontour.io",
+				},
+				Routes: []contourv1.Route{
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-blue",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/blue",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-blue",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/common/exact-blue",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-green",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/common/exact-blue/",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-green",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/blue-exact-green",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-green",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/blue/exact-green",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-default",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/",
+							},
+						},
+					},
+				},
+			},
+		}
+
+		f.CreateHTTPProxyAndWaitFor(serviceProxy, e2e.HTTPProxyValid)
+
+		cases := map[string]string{
```

Comment:
nit: I think this looks correct, but it's not super intuitive to read

not a huge deal but would be nice to either add some comments above some of the cases or use naming of the routes/services that make it clearer

Reply:
Can agree to that, it is just the nature of a few cases that I could not find better naming. Have tried something with comments, not sure how much value that adds though.